rust/connlib/tunnel/src/dns.rs

```diff
@@ -620,6 +648,35 @@ mod tests {
         assert_eq!(match_domain(&domain("a.b.baz.com"), &resources), None);
     }
 
+    #[test_case("**.example.com", "example.com"; "double star matches root domain")]
+    #[test_case("app.**.example.com", "app.bar.foo.example.com"; "double star matches multiple levels within domain")]
+    #[test_case("**.example.com", "foo.example.com"; "double star matches one level")]
+    #[test_case("**.example.com", "foo.bar.example.com"; "double star matches two levels")]
+    #[test_case("*.example.com", "foo.example.com"; "single star matches one level")]
+    #[test_case("*.example.com", "example.com"; "single star matches root domain")]
+    #[test_case("foo.*.example.com", "foo.bar.example.com"; "single star matches one domain within domain")]
+    #[test_case("app.*.*.example.com", "app.foo.bar.example.com"; "single star can appear on multiple levels")]
+    #[test_case("app.f??.example.com", "app.foo.example.com"; "question mark matches one letter")]
+    fn domain_pattern_matches(pattern: &str, domain: &str) {
+        let pattern = Pattern::new(pattern).unwrap();
+
+        let matches = pattern.matches(domain);
+
+        assert!(matches);
+    }
+
+    #[test_case("app.*.example.com", "app.foo.bar.example.com"; "single star does not match two level")]
+    #[test_case("app.*com", "app.foo.com"; "single star does not match dot")]
+    #[test_case("app.**com", "app.foo.com"; "double star does not match dot")]
```

Comment:
This is currently the only case that doesn't pass. I just made this rule up because it felt like it makes sense but it is not how (unix) globbing typically works. Should we move this test case to the expected ones or do we want to special case this?